### PR TITLE
Fix wrong narration when legend selected

### PR DESCRIPTION
**Pull request by krkshitij**
## Current Behavior

Narrator announces highlighted chart element label for unhighlighted chart elements when there is a selected legend

## New Behavior

Narrator announces right label for focused chart element when there is a selected legend

### CODE REVIEW

File: packages/react-charting/src/components/DonutChart/Arc/Arc.tsx [hunk 1]
```diff
@@ -86,9 +86,7 @@ export class Arc extends React.Component<IArcProps, IArcState> {
 
   private _hoverOn(data: IChartDataPoint, mouseEvent: React.MouseEvent<SVGPathElement>): void {
     mouseEvent.persist();
-    if (this.props.activeArc === this.props.data!.data.legend || this.props.activeArc === '') {
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/DonutChart/Arc/Arc.tsx:89]: Why did you remove these checks

File: packages/react-charting/src/components/StackedBarChart/StackedBarChart.base.tsx [hunk 2]
```diff
@@ -317,26 +320,23 @@ export class StackedBarChartBase extends React.Component<IStackedBarChartProps,
   }
 
   private _onBarFocus(pointData: number, color: string, point: IChartDataPoint): void {
-    if (
-      this.state.isLegendSelected === false ||
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/StackedBarChart/StackedBarChart.base.tsx:321]: Can you explain how setting the state everytime helps.

File: packages/react-charting/src/components/DonutChart/Arc/Arc.tsx (hunk 1 above)
Comment (krkshitij) [reply packages/react-charting/src/components/DonutChart/Arc/Arc.tsx:89]: These checks were not completely removed but used in the setState function call to update the visibility of the callout

File: packages/react-charting/src/components/StackedBarChart/StackedBarChart.base.tsx (hunk 2 above)
Comment (krkshitij) [reply packages/react-charting/src/components/StackedBarChart/StackedBarChart.base.tsx:321]: Consider this case:

image

Before the change:
When the 3rd (highlighted) bar is focused, the above condition evaluates to true. The setState function is called and the callout becomes visible with details of the 3rd bar. Now when the 4th (unhighlighted) bar is focused, the condition evaluates to false. The setState function is not called and the callout stays visible with details of the previously focused 3rd bar. So the callout's outdated inner text is used for narration.

After the change:
The setState function is called every time a bar is focused. The callout becomes visible when a highlighted bar is focused and hides when an unhighlighted bar is focused. The callout is always updated with details of the focused bar, which results in correct narration.

**Review by AtishayMsft — APPROVED** [review]
